### ISSUE TITLE
Run developer-uploaded images through pngcrush/optipng/jpegoptim

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#7406]
We load a lot of images we don't control, and we should make sure they are as optimized as they can. This means running them through lossless compression tools like `pngcrush`, `optipng`, and `jpegoptim`. We can save a lot by doing this.

We did this in Marketplace a while ago, and it did help quite a bit. The main caveat was that cachebusting needed to take this into account, because it had to happen on a separate task, so we'd resize the image, save the file, then optimize them, overwriting the previous copy. If somehow a client accessed the resized image before the optimize step, we needed to load the newer version. It can work with the naive way we do cachebusting right now, but it would be better with mozilla/addons-server#2659 instead.

**Comment by diox** [comment]
#2659 is now implemented, so when implementing this one needs to be careful about changing the `icon_hash` after crushing. That probably means changing the way `icon_hash` is generated to use the one of the destination icons instead of the original source, otherwise the hash would not necessarily change after crushing...

Also note that previews need to go through this as well, not just icons.

**Comment by diox** [comment]
QA: make sure image uploads across the site still work correctly once this lands on dev.

**Comment by ioanarusiczki** [comment]
Verified this issue on AMO dev with FF58 (Win10)
Checking uploads for images:
- Collections icons
- User profile picture
- Screenshots and icons for add-ons
- Screenshots  and previews for static themes - issue #7610 was filed already and it is still reproducible
Images and previews for Lightweight themes
